### PR TITLE
Created structure for http layer

### PR DESCRIPTION
**Pull request by olenagerasimova**
Related to #20 - created classes structure of go slices and todos for implementation.

### CODE REVIEW

**Comment by g4s8** [comment]
@0crat elect

**Comment by g4s8** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/artipie/go-adapter/issues/29#issuecomment-615387299))

@g4s8 This is what I know about this job in CT2E6TK9B, as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#29](https://github.com/artipie/go-adapter/issues/29) is [in scope](http://www.zerocracy.com/policy.html#14) for 1days
  * The role is `REV`
  * The job is not assigned to anyone
  * The [budget](http://www.zerocracy.com/policy.html#4) is not set yet
  * These users are banned and won't be assigned:
    * @olenagerasimova[/z](https://www.0crat.com/u/olenagerasimova): This user reported the ticket
  * Job [footprint](https://www.0crat.com/footprint/CT2E6TK9B?q=%7Bjob%3A%27gh%3Aartipie%2Fgo-adapter%2329%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

File: pom.xml [hunk 1]
```diff
@@ -123,18 +127,15 @@ SOFTWARE.
...
   <build>
     <plugins>
-      <plugin>
-        <artifactId>maven-compiler-plugin</artifactId>
-        <configuration>
-          <source>1.8</source>
-          <target>1.8</target>
-        </configuration>
-      </plugin>
+<!--      <plugin>-->
```
Comment (Vatavuk) [inline pom.xml:130]: @olenagerasimova why have you commented out this plugin?

File: src/main/java/com/artipie/http/GoSlice.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+     * Origin.
+     */
+    private final Slice origin;
+
+    /**
+     * Ctor.
+     */
+    public GoSlice() {
+        this.origin = new SliceRoute(
+            new SliceRoute.Path(
```
Comment (Vatavuk) [inline src/main/java/com/artipie/http/GoSlice.java:55]: @olenagerasimova too bad that `Path` is not an interface, we could create something like `GetPath` that would take pattern and a slice as arguments and encapsulate all the `RtRule` instantiation and make the code less verbose. However we can extract a private method that will do the same, if you agree. Something like `Path pathGet(String pattern, Slice slice)`.

File: src/main/java/com/artipie/http/GoSlice.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+import com.artipie.http.rt.SliceRoute;
+import com.artipie.http.slice.LoggingSlice;
+import com.artipie.http.slice.SliceSimple;
+import java.nio.ByteBuffer;
+import java.util.Map;
+import java.util.regex.Pattern;
+import org.reactivestreams.Publisher;
+
+/**
+ * Go slice.
```
Comment (Vatavuk) [inline src/main/java/com/artipie/http/GoSlice.java:39]: @olenagerasimova let's expand the description. `Slice implementation that provides HTTP API for Golang repository.`

**Comment by Vatavuk** [comment]
@olenagerasimova there are confilcts

**Comment by olenagerasimova** [comment]
@Vatavuk thanks, corrected and resolved conflicts

**Comment by 0crat** [comment]
There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/farm/issues):

```
PID: 11@8362b1b91be7, thread: PQ-CT2E6TK9B
com.zerocracy.farm.ruled.RdRules[130] java.lang.IllegalStateException: http://datum.zerocracy.com/latest/rules/pm/no-lost-estimates.xsl says: The estimate for job "gh:artipie/maven-adapter#74" exists in estimates.xml, but the job is absent in the WBS among 52 jobs.
```

[0.53.9](https://github.com/zerocracy/farm): CID: [b2995f6f-54d1-44f8-b9a1-35234afc7791](https://www.0crat.com/footprint/CT2E6TK9B/b2995f6f-54d1-44f8-b9a1-35234afc7791), Type: "Finish order"

**Comment by g4s8** [comment]
@0crat out

**Comment by Vatavuk** [comment]
@g4s8 please try again

**Comment by 0crat** [comment]
@vatavuk[/z](https://www.0crat.com/u/vatavuk) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by g4s8** [comment]
@0crat out

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @vatavuk[/z](https://www.0crat.com/u/vatavuk), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@0crat quality good

**Pull request merged**